gradio/blocks.py

```diff
@@ -705,57 +721,59 @@ def postprocess_data(self, fn_index, predictions, state):
         dependency = self.dependencies[fn_index]
 
         if type(predictions) is dict and len(predictions) > 0:
-            predictions = convert_update_dict_to_list(
+            predictions = convert_component_dict_to_list(
                 dependency["outputs"], predictions
             )
 
         if len(dependency["outputs"]) == 1:
             predictions = (predictions,)
 
-        if block_fn.postprocess:
-            output = []
-            for i, output_id in enumerate(dependency["outputs"]):
-                if predictions[i] is components._Keywords.FINISHED_ITERATING:
-                    output.append(None)
-                    break
-                block = self.blocks[output_id]
-                if getattr(block, "stateful", False):
-                    if not utils.is_update(predictions[i]):
-                        state[output_id] = predictions[i]
-                    output.append(None)
-                else:
-                    prediction_value = predictions[i]
-                    if utils.is_update(prediction_value):
-                        output_value = postprocess_update_dict(block, prediction_value)
-                    else:
-                        output_value = block.postprocess(prediction_value)
-                    output.append(output_value)
-
-        else:
-            output = predictions
+        output = []
+        for i, output_id in enumerate(dependency["outputs"]):
+            if predictions[i] is components._Keywords.FINISHED_ITERATING:
+                output.append(None)
+                continue
+            block = self.blocks[output_id]
+            if getattr(block, "stateful", False):
+                if not utils.is_update(predictions[i]):
```

Comment:
Unrelated to your PR but why do we not set the value of a state to be an update?

Reply:
Hmm good point. I suppose that if the update includes a `value` key, we should assign the `value` to the state. 

I think this check was introduced because if a function returns values in a component-dictionary format (e.g. `{textbox: "hi"}`), we create placeholder updates for the rest of the components, and those placeholder components are a singleton dictionary: `{"__type__"="generic_update"}`. And we wouldn't want to assign this dictionary itself to the state. 

Let me make a fix and add a test!

Reply:
Ok actually `gr.State()` doesn't have an `update()` method at the moment, so passing in a dictionary of updates for `gr.State()` is not supported. There's no strong reason to add `update()` when a function can just return the value itself, so I'll leave this out for now.